### ISSUE TITLE
Application redundancy with two nodes

### ISSUE THREAD

**Issue by michel-zimmer** [rancher/k3s#1845]
**Is your feature request related to a problem? Please describe.**
I don't get to deploy k3s with only two hosts and nodes in a way that makes it possible to kill any of the nodes and still serve the application. (If it's not possible to make changes to the deployments at that time, that's totally fine.) 

So far I've tried this with one server and one agent node. (default configuration, containerd, traefic ingress) Even though everything except one of the two `svclb-traefik-…` pods was deployed on the master, the master did not respond to http requests anymore the moment I took the agent node offline.

**Describe the solution you'd like**
If this is just about missing knowledge regarding configuration or node setup maybe someone can help me out here.
If it's not possible I think it's worth an entry in the documentation. (maybe as part of the architecture or in the FAQ) And I'm interested to hear why and if it becomes possible in the future.

**Describe alternatives you've considered**
I could probably buy more hosts, follow the high availability control plane setup with etcd and with the assumption, that applications would be resilient when the control plane is resilient, then it would work of course. (But this is not feasible for my use case. Then I would have to fallback to deploying applications by hand.)
I also considered deploying any of the other storage backends (that can work with two hosts like postgres), deploying two servers (and no agent). But I don't know if that would work and would like a pointer and recommendations for that direction first.

**Additional context**
TL;DR: The goal is that any of the two hosts can be shut down and all applications keep running, because everything is duplicated all the time. (I don't mind if it's not possible to use kubectl at that time.) I would like to know if that is possible and how it is possible and think this is worth an entry in the documentation somewhere.

**Comment by samirsss** [comment]
We're going through the same exercise and its a real challenge to get this.

As a 1st step to get a 2 node k3s working you're left with 2 options for the backend
1) Postgresql or 2) Mysql. 

We started with Postgresql but getting a 2 node cluster on postgres to be reliable was an issue and hence switched recently to mysql. We have issues there too but its better than trying to use postgresql

You'll have to run coredns, svclb and traefik on both the nodes for HA to function. Also install local-dns so that services can do quicker and proper timely dns lookups.

Metallb or using virtual IP using keepalived or something will be needed to make sure you have a single IP for your application.

Having gone through this and still ongoing i'd say be ready for a bunch of trial and errors to make it work the way you want it.

**Comment by michel-zimmer** [comment]
Thank you for your insights, @samirsss 

That's what I feared. I'm still wondering though, why it doesn't seem possible to take down the agent node, while everything being deployed on the server node and still serve from the server node. Anyone got an idea, what could help here? (I was told of a typical k8s deployment, with only one master and if that goes down, all the workers keep serving the applications without a problem.)

**Comment by michel-zimmer** [comment]
I was unable to reproduce the open point / issue so I'm closing this issue for now.

When I've got one master and one agent and take down the agent, then it might take a moment (until the agent is marked as `NonReady`), but then the reconciler jumps into action and everything get's deployed on the master that has been deployed on the agent and works again.
